src/c++/perf_analyzer/genai-perf/tests/test_artifacts.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from argparse import Namespace
+from pathlib import Path
+
+import pytest
+from genai_perf.constants import DEFAULT_ARTIFACT_DIR
+from genai_perf.main import create_artifacts_dirs
+
+
+@pytest.fixture
+def mock_makedirs(mocker):
```

Comment:
Does this need to be called directly in order for the mock to be applied?

Reply:
I checked online and found we can apply patch decorator for mocking. Modified it to using patch decorator.

Reply:
Can you elaborate the question? I didn't understand.

Reply:
Another way to mock is using
mock_makedirs = mocker.patch("os.makedirs")

Reply:
Ah, that was my confusion. This looked like simply a function definition, but I suppose the `@pytest.fixture` above is causing some code to run.

Reply:
I understood your question now. We don't need to explicitly call the mock_makedirs function. `@pytest.fixture` decorator tells pytest that mock_makedirs is a patch function  and pytest will automatically be able to find that it's a patch for os.makedirs.